pkgs/development/tools/rust/rust-analyzer/update.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq nix-prefetch-github nodePackages.node2nix
+set -euo pipefail
+cd "$(dirname "$0")"
+owner=rust-analyzer
+repo=rust-analyzer
+nixpkgs=../../../../..
+
+# Update lsp
+
+rev=$(
+    curl -s "https://api.github.com/repos/$owner/$repo/releases" |
+    jq 'map(select(.prerelease | not)) | .[0].tag_name' --raw-output
+)
+old_rev=$(sed -nE 's/.*\brev = "(.*)".*/\1/p' ./default.nix)
+if grep -q '0000000000000000000000000000000000000000000000000000' ./default.nix; then
+    old_rev='broken'
+fi
+if [[ "$rev" == "$old_rev" ]]; then
+    echo "Up to date: $rev"
+    exit
+fi
+echo "$old_rev -> $rev"
+
+sha256=$(nix-prefetch-github --prefetch "$owner" "$repo" --rev "$rev" | jq '.sha256' --raw-output)
+sed -e "s/rev = \".*\"/rev = \"$rev\"/" \
+    -e "s/sha256 = \".*\"/sha256 = \"$sha256\"/" \
+    -e "s/cargoSha256 = \".*\"/cargoSha256 = \"0000000000000000000000000000000000000000000000000000\"/" \
+    --in-place ./default.nix
+node_src="$(nix-build "$nixpkgs" -A rust-analyzer.src --no-out-link)/editors/code"
+
+# Check vscode compatibility
+req_vscode_ver="$(jq '.engines.vscode' "$node_src/package.json" --raw-output)"
+req_vscode_ver="${req_vscode_ver#^}"
+cur_vscode_ver="$(nix eval --raw -f "$nixpkgs" vscode.version)"
+if [[ "$(nix eval "(builtins.compareVersions \"$req_vscode_ver\" \"$cur_vscode_ver\")")" != "-1" ]]; then
+    echo "vscode $cur_vscode_ver is incompatible with the extension requiring ^$req_vscode_ver"
+    exit 1
+fi
+
+echo "Prebuilding nix"
+cargo_sha256=$({
+    ! nix-build "$nixpkgs" -A rust-analyzer-unwrapped --no-out-link 2>&1
+} | tee /dev/stderr | sed -nE 's/\s*got:\s*sha256:(\w+)/\1/p' | head -n1)
```

Comment:
You can also use [nix-prefetch](https://github.com/msteen/nix-prefetch). This is how it can be used with rust: https://github.com/Mic92/nix-update/blob/master/nix_update/update.py#L53